odoo/addons/base/models/res_users.py

```diff
@@ -474,6 +474,12 @@ def _check_action_id(self):
         action_open_website = self.env.ref('base.action_open_website', raise_if_not_found=False)
         if action_open_website and any(user.action_id.id == action_open_website.id for user in self):
             raise ValidationError(_('The "App Switcher" action cannot be selected as home action.'))
+        for user in self:
+            if user.action_id.type == "ir.actions.client":
+                action = self.env["ir.actions.client"].browse(user.action_id.id)  # magic
```

Comment:
It's magic because I do browse, instead of using `user.action_id` directly.

# Why

`action_id` field is ir.actions.actions`

https://github.com/odoo/odoo/blob/1ab48bd4c89f21691b34f349338405fcbb85dea4/odoo/addons/base/models/res_users.py#L302-L303

The two models are looks as normal base models.

Parent model:
https://github.com/odoo/odoo/blob/1ab48bd4c89f21691b34f349338405fcbb85dea4/odoo/addons/base/models/ir_actions.py#L22-L25

Inherited model:
https://github.com/odoo/odoo/blob/1ab48bd4c89f21691b34f349338405fcbb85dea4/odoo/addons/base/models/ir_actions.py#L820-L825

However, the second class uses the same sequence for ids. Why? Because the tables are created via psql [INHERITS](https://www.postgresql.org/docs/current/tutorial-inheritance.html)

https://github.com/odoo/odoo/blob/1ab48bd4c89f21691b34f349338405fcbb85dea4/odoo/addons/base/data/base_data.sql#L13

# Magic

The explanation seems too big to put it in the code. On the other hand, having no comments at all would make code  too suspicious. We may have some short version of the explanation, but what if some eager devs would like to solve the riddle by themselves ?

Reply:
So, the word "magic" confuses magicians, who knows how it works :smile: 

sudo is added https://github.com/odoo/odoo/compare/11f5f49322fdfa893e9a3c30548024058c2a5aa2..697dea8eb19344d98cd8176bb53c57bc5ba06e55